src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -233,6 +233,8 @@ export class NotebookEditor implements INotebookEditor {
         const kernel = this.kernelProvider.get(this.file);
 
         if (kernel && !this.restartingKernel) {
+            const promise = this.restartKernelInternal(kernel);
```

Comment:
We have already started restarting the kernel here even before displaying the message
Also, if the user clicks `No` or cancels the prmopt, its too late, as we have restarted it.